File: lib/widgets/common/list_section_header.dart
import 'package:flutter/material.dart';
import '../../theme/app_theme.dart';

/// A stylized header row for list sections (e.g., inside ExpansionTiles).
class ListSectionHeader extends StatelessWidget {
  final String title;
  final IconData icon;
  final Color color;
  final EdgeInsetsGeometry padding;
  final double iconSize;
  final double titleSize;

  const ListSectionHeader({
    super.key,
    required this.title,
    required this.icon,
    required this.color,
    this.padding = const EdgeInsets.symmetric(
      horizontal: AppTheme.spaceLarge,
      vertical: AppTheme.spaceSmall,
    ),
    this.iconSize = AppTheme.iconSizeSmall,
    this.titleSize = AppTheme.fsBodyLarge,
  });

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: padding,
      child: Row(
        children: [
          Icon(icon, color: color, size: iconSize),
          const SizedBox(width: AppTheme.spaceSmall),
          Text(
            title,
            style: TextStyle(
              fontSize: titleSize,
              fontWeight: AppTheme.fwBold,
              color: color,
            ),
          ),
        ],
      ),
    );
  }
}

File: app/src/main/java/io/github/benji377/timety/ui/components/common/ListSectionHeader.kt
package io.github.benji377.timety.ui.components.common

import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.TextUnit
import io.github.benji377.timety.ui.theme.AppTheme


/** Row of a tinted icon followed by a bold title, used to head a list or section. */
@Composable
fun ListSectionHeader(
    title: String,
    icon: ImageVector,
    color: Color,
    modifier: Modifier = Modifier,
    padding: PaddingValues = PaddingValues(
        horizontal = AppTheme.spaceLarge,
        vertical = AppTheme.spaceSmall,
    ),
    iconSize: Dp = AppTheme.iconSizeSmall,
    titleSize: TextUnit = AppTheme.fsBodyLarge,
) {
    Row(
        modifier = modifier.padding(padding),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Icon(icon, contentDescription = null, tint = color, modifier = Modifier.size(iconSize))
        Spacer(Modifier.width(AppTheme.spaceSmall))
        Text(
            title,
            fontSize = titleSize,
            fontWeight = AppTheme.fwBold,
            color = color,
        )
    }
}
